src/raftstore/store/snap.rs

```diff
@@ -254,9 +255,12 @@ mod v1 {
     /// `save` method to make them persistent. When saving a crc32 checksum
     /// will be appended to the file end automatically.
     pub struct Snap {
+        mgr: SnapManager,
```

Comment:
@innerr 

Do you meet this design in other projects, wrapping a manager to operate reference? 
If it is a common way, I think it is ok. Otherwise, we should handle it outside.